assayist/client/query.py

```diff
@@ -15,6 +15,31 @@ def set_connection(neo4j_url):
     neomodel.db.set_connection(neo4j_url)
 
 
+def get_container_by_component(component_name, component_version):
+    """Query for builds by component name and version.
+
+    Finds builds that produced container images which include or embed a component with the
+    specified name and version, and any version preceding that version.
+
+    :param str component_name: name of the component to query for
+    :param str component_version: version of the component to query for
+    :return: list of builds IDs
+    :rtype: list
+    """
+    query = """
+        MATCH (c:Component {{canonical_name: '{name}'}}) <-[:SOURCE_FOR]-
+              (:SourceLocation {{canonical_version: '{version}'}}) <-[:UPSTREAM|EMBEDS*0..]-
+              (target_sl:SourceLocation) <-[:BUILT_FROM]- (build:Build)
+        OPTIONAL MATCH (target_sl) <-[:SUPERSEDES*0..]- (sl:SourceLocation)
+        OPTIONAL MATCH (build) <-[:EMBEDS*]- (:Artifact) <-[:EMBEDS]- (target_sl)
```

Comment:
Well first of all this will never match anything because Builds do not have an EMBEDS edge. You need an extra relationship there to link "build" with the embedded artifacts. Secondly it is not clear to me what this is trying to accomplish. I guess "Find additional builds that embed the source location"? If so I'm not sure this does what you'd expect, because "build" is already a defined set of nodes from the MATCH clause. I think this would use that list of Builds and look for relationships that match, but not expand the list (although I could be wrong). I would also appreciate line comments that state the intend of the clause.

Reply:
Yep, the test case was also modelled incorrect to directly link build to the rsyslog source location. Fixed in current version.